src/Molecules/ToastMessage/ToastMessage.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useImperativeHandle, useState } from 'react';
+import { AnimatePresence, motion } from 'framer-motion';
+
+import { Flexbox } from '../..';
+import { BoxContainer, StyledTypography } from './ToastMessage.styled';
+import { Props } from './ToastMessage.types';
+
+const ToastMessages: React.FC<Props> = React.forwardRef(
+  ({ label, icon, linkButton, isVisible, timeout = 6000 }, ref) => {
+    const [visible, setVisibility] = useState<boolean>(isVisible || false);
+    const [mouseEnter, setMouseEnter] = useState<boolean>(false);
```

Comment:
no need to explicitly type the `useState` here as you get the type implicitly by the init value here
```suggestion
    const [visible, setVisibility] = useState(isVisible || false);
    const [mouseEnter, setMouseEnter] = useState(false);
```